### PR TITLE
Integration tests use DevOps MI for Cosmos

### PR DESCRIPTION
**Pull request by brendankowitz**
## Description
PR and CI pipelines should use MI for Integration Tests

## Related issues
Addresses [AB#126615](https://microsofthealth.visualstudio.com/f8da5110-49b1-4e9f-9022-2f58b6124ff9/_workitems/edit/126615)

## Testing
Existing tests should pass, using new MI credentials for Cosmos

## FHIR Team Checklist
- **Update the title** of the PR to be succinct and less than 65 characters
- **Add a milestone** to the PR for the sprint that it is merged (i.e. add S47)
- Tag the PR with the type of update: **Bug**, **Build**, **Dependencies**, **Enhancement**, **New-Feature** or **Documentation**
- Tag the PR with **Open source**, **Azure API for FHIR** (CosmosDB or common code) or **Azure Healthcare APIs** (SQL or common code) to specify where this change is intended to be released.
- Tag the PR with **Schema Version backward compatible** or **Schema Version backward incompatible** or **Schema Version unchanged** if this adds or updates Sql script which is/is not backward compatible with the code.
- [ ] CI is green before merge [Build Status](https://microsofthealthoss.visualstudio.com/FhirServer/_build/latest?definitionId=27&branchName=main) 
- Review [squash-merge requirements](https://github.com/microsoft/fhir-server/blob/master/SquashMergeRequirements.md)

### Semver Change ([docs](https://github.com/microsoft/fhir-server/blob/master/docs/Versioning.md))
Patch|Skip|Feature|Breaking (reason)

### CODE REVIEW

File: build/jobs/provision-deploy.yml [hunk 1]
```diff
@@ -128,6 +128,28 @@ jobs:
...
+        if("${{ parameters.sql }}" -eq "false"){
+            Write-Host "Add DevOps MI permission to Cosmos databases"
+
+            $principalId = (Get-AzContext).Account.Id
+
+            New-AzCosmosDBSqlRoleAssignment `
+              -AccountName $webAppName `
+              -ResourceGroupName $resourceGroupName `
+              -Scope "/" `
+              -PrincipalId $principalId `
```
Comment (brendankowitz) [inline build/jobs/provision-deploy.yml:141]: Currently failing here since the Federated Principal doesn't exist in the same tenant as the cosmos db

Comment (brendankowitz) [reply build/jobs/provision-deploy.yml:142]: Solved, thanks @mikaelweave :)

File: samples/templates/default-azuredeploy-docker.json [hunk 2]
```diff
@@ -692,20 +693,6 @@
                 "[resourceId('Microsoft.DocumentDb/databaseAccounts', variables('serviceName'))]"
             ]
         },
-        {
-            "condition": "[equals(parameters('solutionType'),'FhirServerCosmosDB')]",
-            "type": "Microsoft.KeyVault/vaults/secrets",
```
Comment (mikaelweave) [inline samples/templates/default-azuredeploy-docker.json:697]: NIT: Do we even need the keyvault anymore? Could be a new story.

File: src/Microsoft.Health.Fhir.CosmosDb/Features/Storage/FhirCosmosClientInitializer.cs [hunk 3]
```diff
@@ -39,16 +40,19 @@ public FhirCosmosClientInitializer(
             ICosmosClientTestProvider testProvider,
             Func<IEnumerable<RequestHandler>> requestHandlerFactory,
             RetryExceptionPolicyFactory retryExceptionPolicyFactory,
+            Lazy<TokenCredential> tokenCredential,
```
Comment (mikaelweave) [inline src/Microsoft.Health.Fhir.CosmosDb/Features/Storage/FhirCosmosClientInitializer.cs:43]: Does this need to be Lazy? I think the implementations of `TokenCredential` already are lazy and the wait is when the token is requested.

File: test/Microsoft.Health.Fhir.Shared.Tests.Integration/Persistence/CosmosDbFhirStorageTestsFixture.cs [hunk 4]
```diff
@@ -79,18 +82,20 @@ public CosmosDbFhirStorageTestsFixture()
         {
             _cosmosDataStoreConfiguration = new CosmosDataStoreConfiguration
             {
-                Host = Environment.GetEnvironmentVariable("CosmosDb:Host") ?? CosmosDbLocalEmulator.Host,
-                Key = Environment.GetEnvironmentVariable("CosmosDb:Key") ?? CosmosDbLocalEmulator.Key,
-                DatabaseId = Environment.GetEnvironmentVariable("CosmosDb:DatabaseId") ?? "FhirTests",
+                Host = Environment.GetEnvironmentVariable("CosmosDb__Host") ?? CosmosDbLocalEmulator.Host,
+                Key = Environment.GetEnvironmentVariable("CosmosDb__Key") ?? CosmosDbLocalEmulator.Key,
```
Comment (mikaelweave) [inline test/Microsoft.Health.Fhir.Shared.Tests.Integration/Persistence/CosmosDbFhirStorageTestsFixture.cs:86]: NIT: Should key be null if we are using MI?

**Review by mikaelweave — APPROVED** [review]
Some small comments but approved. Nice work. 🦾

File: src/Microsoft.Health.Fhir.CosmosDb/Features/Storage/FhirCosmosClientInitializer.cs (hunk 3 above)
Comment (brendankowitz) [reply src/Microsoft.Health.Fhir.CosmosDb/Features/Storage/FhirCosmosClientInitializer.cs:43]: ...good point, I was thinking to not resolve it at all if the key is still being used. Not sure I have a strong opinion on this

File: samples/templates/default-azuredeploy-docker.json (hunk 2 above)
Comment (brendankowitz) [reply samples/templates/default-azuredeploy-docker.json:697]: SQL is still using the keyvault too, it is at least consisntent. Might merge this first so I can mitigate having localauth enabled

**Pull request merged**